platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -288,19 +268,10 @@ public JavaExec systemProperty(String name, Object value) {
      * {@inheritDoc}
      */
     @Override
-    @ToBeReplacedByLazyProperty
```

Comment:
❓ Don't we need `@ReplacesEagerProperty` here? Is that annotation inherited where `@ToBeReplacedByLazyProperty` isn't?

I think for me it would make sense to enforce the annotation to be present on overridden methods as well for documentation purposes.

Reply:
For interception we don't need it: if parent method is annotated, then we don't need to upgrade child method.

For documentation purposes we can add it. I wonder if it adds much of an value? We have also other annotations that are not needed on overriden methods, e.g. `@Input`.

Reply:
Yes so `@ReplacesEagerProperty` is not needed here since parent has this annotation. If we want to support "documentation-only case" we can, we just don't add at the moment